### PR TITLE
provide _NIO1APIShims

### PR DESCRIPTION
**Pull request by weissi**
Motivation:

For package authors, the NIO1 to 2 transition can be taxing because we
keep changing things. So far, the community has done an incredible job
keeping up with our changes but we'd like to make it as easy as possible
continuing to do so.

Modifications:

- add a `_NIO1APIShims` module that when imported tries to provide some
  of the NIO1 API in NIO2 with helpful fixup messages

Result:

easier transition from NIO1 to NIO2

### CODE REVIEW

**Comment by weissi** [comment]
needs to also have #760's renames

**Comment by ianpartridge** [comment]
Well you guys are just the nicest. Thanks.

File: Sources/_NIO1APIShims/NIO1APIShims.swift [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    }
+}
+
+extension EventLoop {
+    @available(*, deprecated, renamed: "makePromise()")
+    public func newPromise<T>(of type: T.Type = T.self, file: StaticString = #file, line: UInt = #line) -> EventLoopPromise<T> {
+        return self.makePromise(of: type, file: file, line: line)
+    }
+    
+    @available(*, deprecated, renamed: "makeSucceededFuture()")
```
[the remark this answers is not in the document]
Comment (weissi) [reply Sources/_NIO1APIShims/NIO1APIShims.swift]: I did and it works but we don't actually need the parens, so removing them.

File: Sources/_NIO1APIShims/NIO1APIShims.swift [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+extension EventLoopGroup {
+    @available(*, deprecated, message: "makeIterator is now required")
+    public func makeIterator() -> NIO.EventLoopIterator {
+        return .init([])
+    }
+}
+
+extension MarkedCircularBuffer {
+    @available(*, deprecated, message: "hasMark is now a property, remove `()`")
```
[the remark this answers is not in the document]
Comment (weissi) [reply Sources/_NIO1APIShims/NIO1APIShims.swift]: there's no `replace`, is there?

File: Sources/_NIO1APIShims/NIO1APIShims.swift [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+public enum ChannelLifecycleError {
+    @available(*, deprecated, message: "ChannelLifecycleError values are now available on ChannelError")
+    public static var inappropriateOperationForState: ChannelError {
+        return ChannelError.inappropriateOperationForState
+    }
+
+}
+
+public enum MulticastError {
+    @available(*, deprecated, message: "MulticastError values are now available on ChannelError")
```
[the remark this answers is not in the document]
Comment (weissi) [reply Sources/_NIO1APIShims/NIO1APIShims.swift]: @Lukasa good point, I added on the type that it has been renamed.

**Comment by weissi** [comment]
blocked by #765

**Pull request merged**